code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -212,14 +212,24 @@
 	become_hearing_sensitive(trait_source = ROUNDSTART_TRAIT)
 	update_step_speed()
 
-/obj/vehicle/sealed/mecha/Destroy()
+//separate proc so that the ejection mechanism can be easily triggered by other things, such as admins
+/obj/vehicle/sealed/mecha/proc/Eject()
 	for(var/M in occupants)
 		var/mob/living/occupant = M
 		if(isAI(occupant))
 			occupant.gib() //No wreck, no AI to recover
 		else
+			occupant.Stun(2 SECONDS)
+			occupant.Knockdown(destruction_knockdown_duration)
+			occupant.throwing = TRUE //This is somewhat hacky, but is the best option available to avoid chasm detection for the split second between the next two lines
 			occupant.forceMove(loc)
```

Comment:
```suggestion
			occupant.forceMove(get_turf(loc))
```

Reply:
also, I think there should be a failsafe code when they get a nullturf

Reply:
Never a bad idea to have fail-safes